src/main/scala/scalismo/ui/rendering/actor/TetrahedralActor.scala

```diff
@@ -0,0 +1,226 @@
+      case _2d: ViewportPanel2D => Some(new TetrahedralMeshActor2D(renderable, _2d))
+    }
+  }
+}
+
+object ScalarTetrahedralMeshFieldActor extends SimpleActorsFactory[ScalarTetrahedralMeshFieldNode] {
+
+  override def actorsFor(renderable: ScalarTetrahedralMeshFieldNode, viewport: ViewportPanel): Option[Actors] = {
+    viewport match {
+      case _: ViewportPanel3D => Some(new ScalarTetrahedralMeshFieldActor3D(renderable))
+      case _2d: ViewportPanel2D => Some(new ScalarTetrahedralMeshFieldActor2D(renderable, _2d))
+    }
+  }
+}
+
+object TetrahedralActor {
+
+  trait TetrahedralRenderable {
+
+    type MeshType
+
+    def opacity: OpacityProperty
+
+    def lineWidth: LineWidthProperty
+
+    def mesh: MeshType
+
+    def node: SceneNode
+  }
+
+  private[actor] object TetrahedralRenderable {
+
+    class TetrahedralMeshRenderable(override val node: TetrahedralMeshNode) extends TetrahedralRenderable {
+
+      type MeshType = TetrahedralMesh[_3D]
+
+      override def mesh: TetrahedralMesh[_3D] = node.transformedSource
+
+      override def opacity: OpacityProperty = node.opacity
+
+      override def lineWidth: LineWidthProperty = node.lineWidth
+
+      def color: ColorProperty = node.color
+    }
+
+    class ScalarTetrahedralMeshFieldRenderable(override val node: ScalarTetrahedralMeshFieldNode) extends TetrahedralRenderable {
+
+      type MeshType = TetrahedralMesh[_3D]
+
+      override def opacity: OpacityProperty = node.opacity
+
+      override def lineWidth: LineWidthProperty = node.lineWidth
+
+      override def mesh: TetrahedralMesh[_3D] = field.mesh
+
+      def scalarRange: ScalarRangeProperty = node.scalarRange
+
+      def field: ScalarVolumeMeshField[Float] = node.transformedSource
+    }
+
+    def apply(source: TetrahedralMeshNode): TetrahedralMeshRenderable = new TetrahedralMeshRenderable(source)
+
+    def apply(source: ScalarTetrahedralMeshFieldNode): ScalarTetrahedralMeshFieldRenderable = new ScalarTetrahedralMeshFieldRenderable(source)
+  }
+
+}
+
+trait TetrahedralActor[R <: TetrahedralRenderable] extends SingleDataSetActor with ActorOpacity with ActorSceneNode {
+  def renderable: R
+
+  override def opacity: OpacityProperty = renderable.opacity
+
+  override def sceneNode: SceneNode = renderable.node
+
+  protected def meshToUnstructuredGrid(template: Option[vtkUnstructuredGrid]): vtkUnstructuredGrid
+
+  protected var unstructuredgrid: vtkUnstructuredGrid = meshToUnstructuredGrid(None)
+
+  // this is invoked from within the rerender method, if the geometry has changed.
+  protected def onGeometryChanged(): Unit
+
+  protected def rerender(geometryChanged: Boolean): Unit = {
+    if (geometryChanged) {
+      unstructuredgrid = meshToUnstructuredGrid(None)
+      onGeometryChanged()
+    }
+
+    actorChanged(geometryChanged)
+  }
+
+  protected def onInstantiated(): Unit = {}
+
+  //FIXME: pick control -- this should probably go into a trait or something.
```

Comment:
Is this something you still want to fix?

Reply:
This seems to be a direct copy from MeshActor where the FIXME originates from.